### ISSUE TITLE
EXPERIMENTAL HiDPI settings render dialog options un-readable

### ISSUE THREAD

**Issue by losmurfs** [eclipse-platform/eclipse.platform.swt#1704]
Latest version of Eclipse 2024-12 with all of the updates. 4 4k displays. Windows 11. Scale: 150% 150% 150% 250% (primary display).
Some dialogs can be resized to compensate for the scaling issues but many can not be resized leaving much dialog content unviewable.
![Screenshot 2024-12-10 102908](https://github.com/user-attachments/assets/a27b0b68-9267-450e-88a2-86d6c6416cd7)
![Screenshot 2024-12-10 102843](https://github.com/user-attachments/assets/4a08bbc3-82ff-4010-bc27-f225cad6a11a)
![Screenshot 2024-12-10 101906](https://github.com/user-attachments/assets/7e978f5d-74e2-49d2-854d-f322a948d7a6)
![Screenshot 2024-12-06 082404](https://github.com/user-attachments/assets/bad08f36-11ab-42f3-9091-9108a723ac0b)

**Comment by HeikoKlare** [comment]
> Did you set your primary display's scale to a scale higher than the display that Eclipse is on? I think that is the key: Windows 11 Primary Display Scale set to 250% and the display that Eclipse is on set to a lower scale like 150%.

Yes, that's exactly the configuration I tried. Unfortunately, I am still not able to reproduce the issue. I moved the issue to the SWT repository, as it is caused by SWT, and assigned the HiDPI label for better tracking.

**Comment by HeikoKlare** [comment]
Closing this as currently the initially reported issue is not reproducible and several improvements in this area have been made since the original report.
